### ISSUE TITLE
Pin_all_from in subdirectories

### ISSUE THREAD

**Issue by Bramjetten** [rails/importmap-rails#6]
When using `pin_all_from`, it's currently impossible to have a module name that's on a different folder level than the `module_path`. 

Example:
My gem contains namespaced Stimulus controllers in `app/assets/javascripts/spina/controllers`. I want those to be available with the module_name `controllers/[some-stimulus-controller].js`, pointing to `/assets/spina/controllers/[some-stimulus-controller].js` so that stimulus-autoloader picks them up.

I've currently worked around this by adding a `prepend_path` attribute to the method (and MappedDir) and simply prepending it to the path in `expand_directories_into(paths)`:

```ruby
Spina.config.importmap.draw do
  pin "@spina/spina", to: "spina.js"
  pin_all_from Spina::Engine.root.join("app/assets/javascripts/spina"), prepend_path: "spina"
end
```

```ruby
def pin_all_from(path, under: nil, preload: false, prepend_path: nil)
  @directories[path] = MappedDir.new(path: path, under: under, preload: preload, prepend_path: prepend_path)
end
```

```ruby
MappedDir  = Struct.new(:path, :under, :preload, :prepend_path, keyword_init: true)
```

```ruby
if mapping.prepend_path
  module_path = File.join(mapping.prepend_path, module_path).to_s
end
```

Is there a better way to fix this? My other two options would be to either a) `pin` every file separately (don't want to do that) or b) create my own stimulus-autoloader, specifically for Spina's Stimulus controllers.

**Comment by dhh** [comment]
I'm not 100% sure I understand the problem. Why isn't `under:` working for you? Could you not mount with `under: "controllers/spina"`, and then pass that absolute path as you do?

**Comment by Bramjetten** [comment]
No, because it uses `absolute_path.basename` to create the `module_path`, resulting in:
`Importmap skipped missing path: controllers/confetti_controller.js`, which should instead be: `spina/controllers/confetti_controller.js`.

On version 0.2.2 (with `append_base_path`) the `filename.relative_path_from(absolute_path)` would generate `controllers/confetti_controller.js` which would then be joined with `absolute_path.basename` (`spina`) correctly. I believe this was more of a coincidence though. This would only work for `pin_all_from Spina::Engine.root.join("app/assets/javascripts/spina")`.

**Comment by dhh** [comment]
Fixed via 1c00c0017922763c49c2fe72cbe4e18df327e874 👍

**Comment by Bramjetten** [comment]
Way better, thank you very much!
